matchzoo/models/dssm_model.py

```diff
@@ -0,0 +1,88 @@
+"""An implementation of DSSM, Deep Structured Semantic Model."""
+from matchzoo import engine
+from keras.models import Model
+from keras.layers import Dense, Input, Dot
+
+
+class DssmModel(engine.BaseModel):
+    """
+    Deep structured semantic model.
+
+    Examples:
+        >>> model = DssmModel()
+        >>> model.params['lr'] = 0.2
+        >>> model.guess_and_fill_missing_params()
+        >>> if model.all_params_filled():
+        ...     model.build()
+
+    """
+
+    @classmethod
+    def get_default_params(cls) -> engine.ParamTable:
+        """:return: model default parameters."""
+        params = super().get_default_params()
+        params.add(engine.Param('w_initializer', 'glorot_normal'))
+        params.add(engine.Param('b_initializer', 'zeros'))
+        # TODO GET TRI-LETTER DIMENSIONALITY FROM FIT-TRANSFORM AS INPUT SHAPE
+        params.add(engine.Param('input_shapes', [(30000,), (30000,)]))
+        params.add(engine.Param('dim_fan_out', 128))
+        params.add(engine.Param('dim_hidden', 300))
+        params.add(engine.Param('activation_hidden', 'tanh'))
+        params.add(engine.Param('activation_prediction', 'softmax'))
+        params.add(engine.Param('lr', 0.01))
```

Comment:
I am thinking about relaxing the constraint of "using only native data types or matchzoo data types" for params. This way, learning rate is no longer a param itself. It is and it should be one of the parameters of `params['optimizer']`. So when the user really wants to tweaking learning rate, the user can do `params['optimizer'] = keras.optimizers.SGD(learning_rate=0.01)` or something. It's not friendly to parameter tuning though, but I'll figure that out separately.

Reply:
ok let's discuss later